File: Classwork/mobileApp/src/anonymousClassesDemos/Demo2.java
package anonymousClassesDemos;

import javax.swing.*;
import java.awt.event.KeyEvent;
import java.awt.event.KeyListener;

import static javax.swing.JFrame.EXIT_ON_CLOSE;

public class Demo2 extends JPanel {
	private Demo2() {
		var frame = new JFrame();
		frame.add(this);
		frame.setVisible(true);
		frame.setSize(800, 400);
		setFocusable(true);
		addKeyListener(new KeyPushingThing());
		frame.setDefaultCloseOperation(EXIT_ON_CLOSE);
	}

	public static void main(String... args) {
		new Demo2();
	}

	public static class KeyPushingThing implements KeyListener {
		@Override
		public void keyTyped(KeyEvent e) {
			System.out.println("3 " + e.getKeyChar());
		}

		@Override
		public void keyPressed(KeyEvent e) {
			System.out.println("2 " + e.getKeyChar());
		}

		@Override
		public void keyReleased(KeyEvent e) {
			System.out.println("1 " + e.getKeyChar());
		}
	}
}

File: Classwork/mobileApp/src/anonymousClassesDemos/Demo2.kt
package anonymousClassesDemos

import java.awt.event.KeyEvent
import java.awt.event.KeyListener
import javax.swing.JFrame
import javax.swing.JPanel
import javax.swing.WindowConstants.EXIT_ON_CLOSE

class Demo2 private constructor() : JPanel() {
	class KeyPushingThing : KeyListener {
		override fun keyTyped(e: KeyEvent) {
			println("3 " + e.keyChar)
		}

		override fun keyPressed(e: KeyEvent) {
			println("2 " + e.keyChar)
		}

		override fun keyReleased(e: KeyEvent) {
			println("1 " + e.keyChar)
		}
	}

	companion object {
		@JvmStatic
		fun main(args: Array<String>) {
			Demo2()
		}
	}

	init {
		val frame = JFrame()
		frame.add(this)
		frame.isVisible = true
		frame.setSize(800, 400)
		isFocusable = true
		addKeyListener(KeyPushingThing())
		frame.defaultCloseOperation = EXIT_ON_CLOSE
	}
}